MobileMiniBrowser should accept the '--url' command line argument

rreno opened the pull request:
#### 31de97a9dff67e49e76d50adaafc58dd663f6ddf
<pre>
MobileMiniBrowser should accept the &apos;--url&apos; command line argument
<a href="https://bugs.webkit.org/show_bug.cgi?id=247378">https://bugs.webkit.org/show_bug.cgi?id=247378</a>
rdar://101878523

Reviewed by Simon Fraser.

This adds support for parsing and acting upon a --url command line arg.
With this patch you would need to edit the scheme in Xcode and pass something like:

--url <a href="https://www.apple.com">https://www.apple.com</a>

to get MobileMiniBrowser to navigate to a specific URL on launch. This is a first step
toward adding support to run-minibrowser so you could launch (Mobile)MiniBrowser like:

run-minibrowser --iphone-simulator --url <a href="https://www.apple.com">https://www.apple.com</a>

* Tools/MobileMiniBrowser/MobileMiniBrowserFramework/WebViewController.m:
(-[WebViewController viewDidLoad]):
(-[WebViewController addProtocolIfNecessary:]):
(-[WebViewController targetURLorDefaultURL]):

Canonical link: <a href="https://commits.webkit.org/256252@main">https://commits.webkit.org/256252@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a030dc39dac5f38b78084db88f69bf16f8108f23

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95213 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/4422 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28134 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104810 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/165069 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99204 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/4466 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/33216 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/87531 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/100705 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100880 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/3253 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/81763 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30223 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85156 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/86994 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73125 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38942 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/18552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36758 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/19835 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4318 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40696 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/81763 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42681 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/39077 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Tools/MobileMiniBrowser/MobileMiniBrowserFramework/WebViewController.m:
```diff
@@ -222,6 +224,36 @@ - (void)removeWebView:(WKWebView *)webView
+    // FIXME: Add support for passing file URLs on the command line.
+    if (targetURLIndex != NSNotFound && targetURLIndex + 1 < [args count]) {
+        NSString *targetURL = [self addProtocolIfNecessary:[args objectAtIndex:targetURLIndex + 1]];
+        NSURL *url = [NSURL URLWithString:targetURL];
```
Might be nicer to say:

```
if (NSURL *url = [NSURL URLWithString: targetURL])
    return url;
```

rreno replied:
I tried that! I know you can do it in C(++). This is what Xcode yells at me with: `Unexpected interface name 'NSURL': expected expression`

I also tried wrapping the assignment in parens and no go.

brentfulgham commented on Tools/MobileMiniBrowser/MobileMiniBrowserFramework/WebViewController.m:
```diff
@@ -222,6 +224,36 @@ - (void)removeWebView:(WKWebView *)webView
     [self.tabViewController.tableView reloadData];
 }
 
+- (NSString *)addProtocolIfNecessary:(NSString *)address
```
It's a shame this implementation can't be shared with normal MiniBrowser.

rreno replied:
Yeah I copy/pasted it directly from Mac MiniBrowser. Maybe the SwiftUI rewrite can unify the code for mac and iOS.

brentfulgham reviewed (approved):
Looks good.

rreno committed:
MobileMiniBrowser should accept the '--url' command line argument
https://bugs.webkit.org/show_bug.cgi?id=247378
rdar://101878523

Reviewed by Simon Fraser.

This adds support for parsing and acting upon a --url command line arg.
With this patch you would need to edit the scheme in Xcode and pass something like:

--url https://www.apple.com

to get MobileMiniBrowser to navigate to a specific URL on launch. This is a first step
toward adding support to run-minibrowser so you could launch (Mobile)MiniBrowser like:

run-minibrowser --iphone-simulator --url https://www.apple.com

* Tools/MobileMiniBrowser/MobileMiniBrowserFramework/WebViewController.m:
(-[WebViewController viewDidLoad]):
(-[WebViewController addProtocolIfNecessary:]):
(-[WebViewController targetURLorDefaultURL]):

Canonical link: https://commits.webkit.org/256252@main
